stl/inc/ranges

```diff
@@ -904,15 +928,15 @@ namespace ranges {
         inline constexpr empty_view<_Ty> empty;
     } // namespace views
 
-    template <copy_constructible _Ty>
-        requires is_object_v<_Ty>
+    template <_Valid_movable_box_object _Ty>
     class single_view : public view_interface<single_view<_Ty>> {
     public:
         // clang-format off
         single_view() requires default_initializable<_Ty> = default;
         // clang-format on
 
         constexpr explicit single_view(const _Ty& _Val_) noexcept(is_nothrow_copy_constructible_v<_Ty>) // strengthened
+            requires(copy_constructible<T>)

```

Comment:
The tests are failing with:

```
In file included from C:\a\1\s\tests\std\tests\P1659R3_ranges_alg_ends_with\test.cpp:7:
D:\build\out\inc\ranges(939,41): error: use of undeclared identifier 'T'
            requires(copy_constructible<T>)
                                        ^
```

Before pushing changes, please always rebuild and run a limited test pass - it is extremely fast to run a small number of directories. Newly added test directories should always be run, plus any other directories of interest. (For `<ranges>` changes, I would suggest running all tests with `*ranges*` in the directory name, which isn't exactly *fast*, but still much faster than a full test pass.)

If you're developing on a laptop, a VM in the internal DevTestLab may be faster for running tests (and avoid tying up your machine while it runs).

Reply:
The issue of "I forgot to rerun the build before rerunning the tests" crops up again -.-